concept/thing/thing_manager.rs

```diff
@@ -806,31 +838,226 @@ impl ThingManager {
             Err(error) => Err(ConceptReadError::SnapshotGet { source: error }),
         }
     }
+
+    pub(crate) fn relation_exists<'a>(
+        &self,
+        snapshot: &impl ReadableSnapshot,
+        relation: &Relation<'a>,
+    ) -> Result<bool, ConceptReadError> {
+        match snapshot.get::<0>(relation.vertex().as_storage_key().as_reference()) {
```

Comment:
I couldn't add all the needed traits impls for `ThingAPI` as it's not a struct (spent some time there, coudn't finish it up), so I just created some special functions for `relation`s alongside `object`s instead of making it accept `&impl ThingAPI`